azure/services/virtualmachines/client.go

```diff
@@ -31,11 +34,35 @@ import (
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
-// AzureClient contains the Azure go-sdk Client.
-type AzureClient struct {
-	virtualmachines compute.VirtualMachinesClient
+type (
+	// Client wraps go-sdk.
+	Client interface {
```

Comment:
Do we need this to interface to be exported? Is it used anywhere outside of this package?

Reply:
Yes, it's used in [scalesetvms.go](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2813/files#diff-afa3aa43d3541f80b10af246e174440ced390fc557b6a907525183d4d9dd5106R51).